src/segment-loader.js

```diff
@@ -590,23 +595,6 @@ export default class SegmentLoader extends videojs.EventTarget {
     }
   }
 
-  createTransmuxer_() {
```

Comment:
This moved to `segmentTransmuxer.createTransmuxer` for a few reasons: 
1. We use it in tests in exactly the same manner
2. We need to add some special properties to the transmuxer when we create it in segmentTransmuxer to prevent state from being kept in the segmentTransmuxer itself
3. We need to clear those same properties on terminate, which we do by wrapping terminate in segmentTransmuxer
4. segmentTransmuxer already handles all the other transmuxer message logic, so having logic here for it is weird.